### PR TITLE
Merge to Master #1

### PR DESCRIPTION
**Pull request by russellwmoore**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  module: {
+    rules: [
+      {
+        test: /jsx?$/,
+        include: resolve(__dirname, './app'),
+        loader: 'babel-loader'
+      },
+      {
+        test: /\.css$/,
```
Comment (russellwmoore) [inline webpack.config.js]: Take out the CSS loader if you are not importing css into react directly

File: db/models/product.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        validate: {
+            len: [0, 500],
+        }
+    },
+    inventory: {
+        type: Sequelize.INTEGER,
+        defaultValue: 5000,
+        allowNull: false,
+        validate: {
+            isNumeric: true,
```
Comment (russellwmoore) [inline db/models/product.js]: Is this a thing?

File: db/models/product.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            min: 0,
+            max: 5000,
+        }
+    },
+    price: {
+        type: Sequelize.DECIMAL(6,2),
+        allowNull: false,
+        validate: {
+            isNumeric: true,
```
Comment (russellwmoore) [inline db/models/product.js]: Is this a thing?

File: db/models/shippingaddress.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      isUppercase: true,
+    }
+  },
+  zip: {
+    type: Sequelize.INTEGER,
+    allowNull: false,
+    validate: {
+      notEmpty: true,
+      isInt: true,
+      len: [5]
```
Comment (russellwmoore) [inline db/models/shippingaddress.js]: Is this an array? It might be! Check it??

File: db/models/user.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    },
+  },
+  password: {
+    type: Sequelize.TEXT,
+    allowNull: false,
+    validate: {
+      notEmpty: true,
+    },
+  },
+  sessionId: {
```
Comment (russellwmoore) [inline db/models/user.js]: This info might belong in a model that represents the fact that we are already aware of non-signed up users. 
Using a session/UUID/UserId combo.

File: main.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const chalk = require('chalk');
+const { connection } = require('./db');
+const server = require('./server');
+
+const PORT = 3000;
```
Comment (russellwmoore) [inline main.js]: .env variables? How are we going to deploy AND keep some things secret...

File: package.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "TEAM-A",
+  "version": "1.0.0",
+  "description": "This is our Readme file. Let's build something!",
+  "main": "main.js",
+  "scripts": {
+    "start-dev": "npm run build-watch & npm run start-server",
```
Comment (russellwmoore) [inline package.json]: be sure to add your desired environment here. Dev or prod?

File: package.json [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    "start-server": "nodemon main.js -e html,js,css --ignore public",
+    "build-watch": "webpack -w",
```
Comment (russellwmoore) [inline package.json]: missing a build script for your deploy. Heorku needs a build script to know how to read good and build your app. ie webpack

File: package.json [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    "babel-preset-react": "^6.24.1",
+    "bcrypt": "^3.0.7",
+    "chalk": "^3.0.0",
+    "cjs": "0.0.11",
+    "cookie-parser": "^1.4.4",
+    "date-fns": "^2.9.0",
+    "express": "^4.17.1",
+    "faker": "^4.1.0",
+    "nodemon": "^2.0.2",
+    "npm-run-all": "^4.1.5",
```
Comment (russellwmoore) [inline package.json]: who dis? Do you need it?

File: package.json [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  "homepage": "https://github.com/1909-FLEX-GRACE-SHOPPER/TEAM-A#readme",
+  "dependencies": {
+    "@date-io/date-fns": "^2.0.1",
+    "@material-ui/core": "^4.9.0",
+    "@material-ui/icons": "^4.5.1",
+    "@material-ui/pickers": "^3.2.9",
+    "@material-ui/styles": "^4.9.0",
+    "axios": "^0.19.1",
+    "babel": "^6.23.0",
```
Comment (russellwmoore) [inline package.json]: all babel to dev-dep

File: package.json [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    "nodemon": "^2.0.2",
+    "npm-run-all": "^4.1.5",
+    "path": "^0.12.7",
+    "pg": "^7.17.1",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
+    "react-redux": "^7.1.3",
+    "react-router-dom": "^5.1.2",
+    "redux": "^4.0.5",
+    "redux-devtools-extension": "^2.13.8",
```
Comment (russellwmoore) [inline package.json]: dev-dep. HOWEVER -- check how you are actually implementing in your code. You'll probably need to check for a development mode then use this and other dev tools

File: package.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-react": "^6.24.1",
+    "bcrypt": "^3.0.7",
+    "chalk": "^3.0.0",
+    "cjs": "0.0.11",
+    "cookie-parser": "^1.4.4",
+    "date-fns": "^2.9.0",
+    "express": "^4.17.1",
+    "faker": "^4.1.0",
+    "nodemon": "^2.0.2",
```
Comment (russellwmoore) [inline package.json]: dev-dep

File: public/index.html [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<!doctype html>
+<html>
+  <head>
+    <title>Grace Shopper</title>
+    <link rel="stylesheet" href="/style.css" />
+    <link href="https://fonts.googleapis.com/css?family=Montserrat&display=swap" rel="stylesheet">
```
Comment (russellwmoore) [inline public/index.html]: slowing us all down

File: public/style.css [hunk 14]
```diff
@@ -0,0 +1,13 @@
+* {
```
Comment (russellwmoore) [inline public/style.css]: blah

File: seed.js [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+    console.log(chalk.red('Error in seed function'));
+    console.error(e);
+  }
+}
+
+//run the seed function
+seed()
+  .then(() => {
+    console.log(chalk.green('Database sync successful'));
+    connection.close();
```
Comment (russellwmoore) [inline seed.js]: I think close returns a promise!! so deal with it.

File: server/api/index.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+
+const router = express.Router();
+
+router.use(express.json());
```
Comment (russellwmoore) [inline server/api/index.js]: huh? does this happen earlier in the chain?

File: server/auth/login.js [hunk 17]
```diff
@@ -0,0 +1,43 @@
+var express = require('express');
+var router = express.Router();
+const { User } = require('../../db');
+const { generateSessionId, compare } = require('../utils')
+const moment = require('moment')
+
+router.use(express.json());
```
Comment (russellwmoore) [inline server/auth/login.js]: why have here. earlier up?

File: server/auth/who.js [hunk 18]
```diff
@@ -0,0 +1,13 @@
+const express = require('express');
+const Sequelize = require('sequelize');
+const router = express.Router();
+
```
Comment (russellwmoore) [inline server/auth/who.js]: 👍🏻

File: server/index.js [hunk 19]
```diff
@@ -0,0 +1,59 @@
+const express = require('express');
+const path = require('path');
+const chalk = require('chalk');
+const cookieParser = require('cookie-parser');
+const { User } = require('../db');
+
+const server = express();
+
+//middleware here
+server.use(express.json());
```
Comment (russellwmoore) [inline server/index.js]: delete all other json parsers plz

File: server/index.js [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+server.use((req, res, next) => {
+    if (req.cookies.sessionId) {
+        User.findOne({
+            where: {
+                sessionId: req.cookies.sessionId
+            }
+        })
+        .then(user => {
+            if (user) {
+                req.loggedIn = true;
```
[the remark this answers is not in the document]
Comment (ajmorozoff) [reply server/index.js]: Done

File: server/index.js [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+            next(e);
+        })
+    }
+    else {
+        next();
+    }
+});
+
+//routing here
+server.use('/api', require('./api'));
```
[the remark this answers is not in the document]
Comment (ajmorozoff) [reply server/index.js]: Done

File: server/index.js [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+server.use('/auth', require('./auth'));
+
+//main route to serve index.html
+server.get('/', (req, res) => {
```
[the remark this answers is not in the document]
Comment (ajmorozoff) [reply server/index.js]: Done

File: server/utils.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
+const bcrypt = require('bcrypt')
+
+//utility functions for server
+
+//generate the sessionId
+const generateSessionId = () => {
+  return Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
```
[the remark this answers is not in the document]
Comment (ajmorozoff) [reply server/utils.js]: Done

**Pull request merged**